#!/bin/sh -e
##  by Joost Yervante Damad <joost@lumatec.be>
##
## DP: enable usage of CCACHE

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -ur openmsx-0.5.0.orig/build/main.mk openmsx-0.5.0.orig/build/main.mk
--- openmsx-0.5.0.orig/build/main.mk	2005-01-06 19:53:35.000000000 +0000
+++ openmsx-0.5.0/build/main.mk	2005-01-06 20:18:10.000000000 +0000
@@ -316,6 +316,12 @@
 endif
 export CXX:=$(OPENMSX_CXX)
 
+# Check if ccache usage was requested
+ifeq ($(USE_CCACHE),true)
+	CXX:=ccache $(CXX)
+endif
+
+
 # Use precompiled headers?
 # TODO: Autodetect this: USE_PRECOMPH == compiler_is_g++ && g++_version >= 3.4
 #       In new approach, Make >= 3.80 is needed for precompiled header rules.

